### PR TITLE
[CONTINT-3877] Add admission controller webhook for autoscaling

### PR DESCRIPTION
**Pull request by AliDatadog**
### What does this PR do?
This PR adds the admission controller webhook for autoscaling which applies vertical recommendations.

### Motivation
We would like to apply vertical recommendations to new pods.

### Describe how to test/QA your changes
Deploy a dummy application. Create a Datadog Autoscaling Custom resource with vertical scaling recommendations. Deploy it and delete a pod. The new pod should have the resource requests/limits defined in the custom resource.

I tested it with a custom build using a fake recommender:
```
	if ns == "testnamespace" {
		return []datadoghq.DatadogPodAutoscalerContainerResources{
			{
				Name:     "container1",
				Limits:   corev1.ResourceList{"cpu": resource.MustParse("500m")},
				Requests: corev1.ResourceList{"memory": resource.MustParse("256Mi")},
			},
		}, nil
	}
```

And we can see the recommendations applied to a resource-less pod:
```
    name: container1
    resources:
      limits:
        cpu: 500m
      requests:
        cpu: 500m
        memory: 256Mi
```

### CODE REVIEW

File: pkg/config/setup/config.go [hunk 1]
```diff
@@ -1061,6 +1061,8 @@ func InitConfig(config pkgconfigmodel.Config) {
 	config.BindEnvAndSetDefault("admission_controller.container_registry", "gcr.io/datadoghq")
 	config.BindEnvAndSetDefault("admission_controller.timeout_seconds", 10) // in seconds (see kubernetes/kubernetes#71508)
 	config.BindEnvAndSetDefault("admission_controller.service_name", "datadog-admission-controller")
+	config.BindEnvAndSetDefault("admission_controller.autoscaling.enabled", false)
+	config.BindEnvAndSetDefault("admission_controller.autoscaling.endpoint", "/autoscale")
```
Comment (clamoriniere) [inline pkg/config/setup/config.go:1065]: I don't think we need a parameter to define the endpoint path

Comment (AliDatadog) [reply pkg/config/setup/config.go:1065]: We have one for every other webhook

**Review by louis-cqrl — APPROVED** [review]
LGTM for files ASC owns

File: pkg/clusteragent/admission/mutate/autoscaling/autoscaling.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	if ownerRef.Kind == kubernetes.ReplicaSetKind {
+		ownerRef.Kind = kubernetes.DeploymentKind
+		ownerRef.Name = kubernetes.ParseDeploymentForReplicaSet(ownerRef.Name)
+	}
+	// ParseDeploymentForReplicaSet returns "" when the parsing fails
+	if ownerRef.Name == "" {
+		return false, fmt.Errorf("no owner found for pod %s", pod.Name)
+	}
+
+	recommendations, err := w.recommender.GetRecommendations(pod.Namespace, ownerRef)
```
Comment (AliDatadog) [inline pkg/clusteragent/admission/mutate/autoscaling/autoscaling.go:113]: TODO: handle the `nil, nil` return case

**Pull request closed without merging**